platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -171,137 +230,269 @@ public Schema getSchema(CFMetaData metadata) {
-                    IAppendableColumn col;
-                    // Extracting the value for each column from a single row
-                    for (ColumnData cd : row) {
-                        // Filter the column to only load the one marked true by columnToLoad[]
-                        if (columnToLoad[i]) {
-                            if (cd.column().isSimple()) {
-                                Cell cell = (Cell) cd;
-                                AbstractType<?> cellType = cell.column().cellValueType();
-                                value = cellType.getSerializer().deserialize(cell.value());
-                            } else {
-                                // Hillview won't process the complex data
-                                throw new RuntimeException("Hillview can't convert complex data");
+    /**
+     * Lazy and non-lazy loader will call this function to load specific column
+     * marked by columnToLoad
+     */
+    private static void loadColumns(SSTableReader ssTableReader, List<IAppendableColumn> columns,
+            PartitionColumns columnDefinitions, List<CassandraTokenRange> tokenRanges, String localEndpoint) {
+        int colCount = columnDefinitions.size();
+        ColumnFilter cf = ColumnFilter.selection(columnDefinitions);
+        List<Long> arrPrefixComparison = new ArrayList<>();
+        HashMap<CQL3Type, TypeSerializer<?>> serializers = new HashMap<CQL3Type, TypeSerializer<?>>();
+        Iterator<ColumnDefinition> colIter = columnDefinitions.regulars.simpleColumns();
+        // init reusable serializer and prefixComparison
+        while(colIter.hasNext()) {
+            ColumnDefinition cd = colIter.next();
+            CQL3Type type = cd.type.asCQL3Type();
+            serializers.put(type, type.getType().getSerializer());
+            arrPrefixComparison.add(cd.name.prefixComparison);
+        }
+
+        for (CassandraTokenRange tr : tokenRanges) {
+            // Load the partition if the local endpoint is listed as the first endpoint
+            if (tr.endpoints.get(0).equals(localEndpoint)) {
+                DataRange range = DataRange.forTokenRange(tr.tokenRange);
+                SSTableReadsListener listener = CassandraSSTableLoader.newReadCountUpdater();
+                ISSTableScanner currentScanner = ssTableReader.getScanner(cf, range, false, listener);
+
+                Spliterators.spliteratorUnknownSize(currentScanner, Spliterator.CONCURRENT)
+                    .forEachRemaining((partition) -> {
+                        Unfiltered unfiltered = partition.next();
+                        if (unfiltered instanceof Row) {
+                            Row row = (Row) unfiltered;
+                            int currentColumn = 0;
+                            Object value;
+                            IAppendableColumn col;
+                            // Extracting the value of each column from a single row
+                            for (ColumnData cd : row) {
```

Comment:
instead of iterating over the row, iterate over ColumnDefinition and call row.getCell(col)